### ISSUE TITLE
Output of PyTorch and ONNX Runtime do not match

### ISSUE THREAD

**Issue by salonit** [pytorch/tutorials#1335]
Tried exporting a model from PyTorch to ONNX  but after running it using the ONNX Runtime the outputs do not seem to match.
This is the code snippet that I have used for exporting:

`from torch.autograd import Variable
from models.model_utils import create_model
import torch.onnx
import torchvision
import torch
from config.train_config import parse_train_configs
import onnx
import numpy as np
import onnxruntime

configs =  parse_train_configs()
model = create_model(configs)

dummy_input = torch.randn(1, 3, configs.img_size, configs.img_size, requires_grad = True)
dummy_out = model(dummy_input)
state_dict = torch.load('/home/lrm/repos/Complex-YOLOv4-Pytorch/checkpoints/complexer_yolo/Model_complexer_yolo_epoch_1800.pth')
model.load_state_dict(state_dict)

torch.onnx.export(model, dummy_input, "yolo_1800.onnx", export_params = True,
        opset_version =11, input_names = ['input'],   # the model's input names
        output_names = ['output'], # the model's output names
                  dynamic_axes={'input' : {0 : 'batch_size'},    # variable lenght axes
                                'output' : {0 : 'batch_size'}})
onnx_model = onnx.load('yolo_1800.onnx')
onnx.checker.check_model(onnx_model)

ort_session = onnxruntime.InferenceSession("yolo_1800.onnx")

def to_numpy(tensor):
    return tensor.detach().cpu().numpy() if tensor.requires_grad else tensor.cpu().numpy()

# compute ONNX Runtime output prediction
ort_inputs = {ort_session.get_inputs()[0].name: to_numpy(dummy_input)}
ort_outs = ort_session.run(None, ort_inputs)

# compare ONNX Runtime and PyTorch results
np.testing.assert_allclose(to_numpy(dummy_out), ort_outs[0], rtol=1e-03, atol=1e-05)

print("Exported model has been tested with ONNXRuntime, and the result looks good!")

# Output

Mismatched elements: 42350 / 43320 (97.8%)
Max absolute difference: 168.14859
Max relative difference: 101185.69
 x: array([[[ 1.726743e+01,  1.159815e+01,  7.745127e+00, ...,
          6.100558e-01,  1.621268e-07,  9.623343e-01],
        [ 4.459081e+01,  1.060910e+01,  1.184277e+01, ...,...
 y: array([[[ 1.637772e+01,  1.484044e+01,  7.548301e+00, ...,
          2.327538e-01,  4.627705e-04,  9.783132e-01],
        [ 4.849532e+01,  1.091816e+01,  5.151544e+00, ...,...

`

Both the models can be found [here](https://drive.google.com/file/d/1aXsesHo7UXjD3iuxFpRsNYAHVS3Tf8tG/view?usp=sharing)

Pytorch version: 1.7.1+cu101
onnx version: 1.8.1

**Comment by svekars** [comment]
I'm not sure which tutorial this applies to and this looks like it was a few PyTorch versions ago. Can you try with 2.0 and see if this was resolved? Closing for now, as this might be no longer an issue - feel free to reopen if needed and provide a link to the tutorial that was failing. For questions not related to tutorials, please post in https://dev-discuss.pytorch.org/
